bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/AddressPersistenceService.kt

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.eclipse.tractusx.bpdm.gate.service
+
+import jakarta.transaction.Transactional
+import org.eclipse.tractusx.bpdm.common.util.replace
+import org.eclipse.tractusx.bpdm.gate.api.model.AddressGateInputRequest
+import org.eclipse.tractusx.bpdm.gate.entity.LogisticAddress
+import org.eclipse.tractusx.bpdm.gate.repository.GateAddressRepository
+import org.springframework.stereotype.Service
+
+@Service
+class AddressPersistenceService(private val gateAddressRepository: GateAddressRepository) {
+
+    @Transactional
+    fun persistAddressBP(addresses: Collection<AddressGateInputRequest>) {
+
+        val externalIdColl: MutableCollection<String> = mutableListOf()
+        addresses.forEach { externalIdColl.add(it.externalId) }
+
+        val addressRecord = gateAddressRepository.findByExternalIdIn(externalIdColl)
+
+        addresses.forEach { address ->
+            val fullAddress = address.toAddressGate()
+            addressRecord.find { it.externalId == address.externalId }?.let { existingAddress ->
+                updateAddress(existingAddress, address)
+                gateAddressRepository.save(existingAddress)
+            } ?: run {
+                gateAddressRepository.save(fullAddress)
+            }
+        }
+    }
+
+    private fun updateAddress(address: LogisticAddress, addressRequest: AddressGateInputRequest) {
```

Comment:
We firstly had it that way, but it seems that if we change it to an entity instead of the request DTO, wherever we try to update the persisted data it is unable to find some persisted entities. For example:

 ![image](https://github.com/eclipse-tractusx/bpdm/assets/106146227/5352fb70-8727-49bf-a10b-7e052276976e)

Reply:
ok, we keep it for now